docs/docs/0.5.0/policies/mutual-tls.md

```diff
@@ -1,167 +1,229 @@
 
-A complete workflow of changing CA type from `builtin` to `provided` looks the following way:
+You can retrieve the secrets via `kumactl` on both Universal and Kubernetes, or via `kubectl` on Kubernetes only:
 
-1. A user must first upload a custom Root CA certificate using `kumactl manage ca provided` command, e.g.
+:::: tabs :options="{ useUrlFragment: false }"
+::: tab "kumactl"
 
-   ```shell
-   kumactl manage ca provided certificates add --mesh demo --cert-file demo.root_ca.crt --key-file demo.root_ca.key
-   ```
+The following command can be executed on any Kuma backend:
 
-   ::: warning
-   This is a security-sensitive opreation, which implies very restrictive default settings.
+```sh
+$ kumactl get secrets [-m MESH]
+MESH      NAME                           AGE
+default   default.ca-builtin-cert-ca-1   1m
+default   default.ca-builtin-key-ca-1    1m
+```
+:::
+::: tab "kubectl"
 
-   If you are just Getting Started with Kuma, run `kumactl manage ca provided` commands on the same machine where `kuma-cp` is running.
-   :::
+The following command can be executed only on Kubernetes:
+
+```sh
+$ kubectl get secrets -n kuma-system
+NAME                             TYPE                                  DATA   AGE
+default.ca-builtin-cert-ca-1     system.kuma.io/secret                 1      1m
+default.ca-builtin-key-ca-1      system.kuma.io/secret                 1      1m
+```
+:::
+::::
 
-2. Next, a user must **disable** mTLS on a `Mesh` before he will be allowed to change CA type from `builtin` to `provided`, e.g.
+## Usage of "provided" CA
 
-   On Universal:
+Whenever we want to provide our own CA root certificate and key, we can use the `provided` backend. 
 
-   ```shell
-   echo "
-   type: Mesh
-   name: demo
-   mtls:
-    # there is no enabledBackend property which means mTLS is enabled
-    backends:
-    - name: ca-1
-      type: builtin # use Builtin CA (a unique Root CA certificate will be generated automatically)
-   " | kumactl apply -f -
-   ```
-
-   On Kubernetes:
-
-   ```shell
-   echo "
-   apiVersion: kuma.io/v1alpha1
-   kind: Mesh
-   metadata:
-     name: default
-   spec:
-     mtls:
-      # there is no enabledBackend property which means mTLS is enabled
-      backends:
-      - name: ca-1
-        type: builtin # use Builtin CA (a unique Root CA certificate will be generated automatically)
-   " | kubectl apply -f -
-   ```
+Unlike the `builtin` backend, with `provided` we need to first upload the certificate and key that Kuma will use as [Secret resources](/docs/0.5.0/documentation/secrets), and then reference the Secrets to the mTLS configuration.
 
-3. A user can now change CA type to `provided` and **enable** mTLS back after that
+We are responsible for providing the CA root certificate + key and also to manage their lifecycle. Kuma will then use our CA root certificate + Key to automatically provision (and rotate) data plane certificates for every replica of every service.
```

Comment:
We or You? Both forms are used in this page. We may suggest us - Kuma developers/maintainers.